### PR TITLE
S5T8 & S5T5 Add Contact Frontend & Validation

### PR DESCRIPTION
**Pull request by mallowmew**

### CODE REVIEW

**Comment by mallowmew** [comment]
Thsi should all be done but breaks because the model doesn't have a method yet :)

File: public/js/validations.js [hunk 1]
```diff
@@ -14,4 +14,41 @@ function isTime(time) {
...
+    let regEx = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+    if (regEx.test(email)) {
+        return true
+    } else {
+        return false
+    }
+}
+
+function isUrl(url) {
+    let regEx = /^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$/gm
```
Comment (AlisonYoon) [inline public/js/validations.js]: should <http(s)> be optional or mandatory? On company's name card or something, many places omit http(s) in front of the website url.

File: public/js/validations.js [hunk 2]
```diff
@@ -14,4 +14,41 @@ function isTime(time) {
...
+    if (regEx.test(phone)) {
+        return true
+    } else {
+        return false
+    }
+}
+
+function isEmail(email) {
+    // email regex from http://emailregex.com - "Email Address Regular Expression That 99.99% Works."
+    let regEx = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```
Comment (AlisonYoon) [inline public/js/validations.js]: you can have periods in id, but after @, there should be limit in number of periods isn't it? 
(sdfsdfsdfadf@sdfsdf.dsf.sdfs.sdf.asdf.asdf.asdf.asdf) -> this weird address passes the email validation

File: public/js/validations.js (hunk 1 above)
Comment (mallowmew) [reply public/js/validations.js]: It's optional but we might consider trimming or enforcing it in the backend

File: public/js/validations.js (hunk 2 above)
Comment (mallowmew) [reply public/js/validations.js]: sdfsdfsdfadf@sdfsdf.dsf.sdfs.sdf.asdf.asdf.asdf.asdf could actually be a standard-compliant email address ¯\_(ツ)_/¯

File: public/js/validations.js [hunk 3]
```diff
@@ -14,4 +14,41 @@ function isTime(time) {
     } else {
         return false
     }
+}
+
+function isPhoneNumber(phone) {
+    let regEx = /^(([+][(]?[0-9]{1,3}[)]?)|([(]?[0-9]{4}[)]?))\s*[)]?[-\s\.]?[(]?[0-9]{1,3}[)]?([-\s\.]?[0-9]{3})([-\s\.]?[0-9]{3,4})$/gm
```
Comment (AlisonYoon) [inline public/js/validations.js]: after the country code, if the first digit is 0 then you should omit it, so for example it goes:
+44(0)7100123345
Does this kind of filtering for display be done in this branch do you think?

File: public/js/validations.js (hunk 1 above)
Comment (AlisonYoon) [reply public/js/validations.js]: for now, without http(s), it doesn't pass the regex validation.
for instance something like (www.mayden.com) doesn't pass this.

File: public/css/genericStyles.css [hunk 4]
```diff
@@ -43,7 +43,7 @@ input[type="submit"] {
     border-top: 1px solid #CCCCCC
 }
 
-.moreInfo {
+.more-info {
```
Comment (AlisonYoon) [inline public/css/genericStyles.css]: I think this was done in Matt's branch but margin-top and margin-bottom doesn't have to be in two separate lines. 
margin: 10px 0;
should do the job.

**Pull request merged**